### PR TITLE
Succeed with warnings

### PR DESCRIPTION
**Pull request by ClarenceGardner**
Allow some situations resulting in a file not being synced to be treated as warnings rather than errors.

### CODE REVIEW

File: o4/o4.py [hunk 1]
```diff
@@ -866,8 +894,11 @@ def man(cmd):
     run_cmd(cmd)
 
     actual_cl, _ = get_fstat_cache(changelist)
-    with open('.o4/changelist', 'wt') as fout:
-        print(actual_cl, file=fout)
+    try:
+        os.remove(INCOMPLETE_INDICATOR)
+    except FileNotFoundError:
+        with open(SYNCED_CL_FILE, 'wt') as fout:
+            print(actual_cl, file=fout)
```
Comment (guw) [inline o4/o4.py]: That looks like an anti-pattern - control-flow by exception. Is there a specific reason you are doing `os.path.exists` earlier up in the file but not in this location?

Also, the file is different, i.e. if you fail to delete `INCOMPLETE_INDICATOR` you are just opening `SYNCED_CL_FILE`.  Is it guaranteed to exist at this point?

File: o4/o4.py [hunk 2]
```diff
@@ -955,21 +986,49 @@ def move_except(from_dir, to_dir, but_not):
...
+        err_print('\n'.join(sorted(passthroughs)))
+        err_print(ftr)
+        if not files:
+            with open(INCOMPLETE_INDICATOR, 'w') as f:
+                pass
+            err_print(f'{CLR} o4 IS SUCCEEDING EVEN THOUGH SOME FILES ARE NOT UP-TO-DATE')
+            err_print(ftr)
+            sys.exit(0)
+
+    sys.exit(f"{CLR}*** ERROR: Pipeline ended with {n} fstat lines.")
```
Comment (guw) [inline o4/o4.py]: This is unrelated to the code change something for a TODO list. Is there a better error message than this technical description?

File: o4/o4.py (hunk 1 above)
Comment (ClarenceGardner) [reply o4/o4.py]: The SYNCED file is being created if there was no INCOMPLETE file. The only purpose of INCOMPLETE is to prevent SYNCED (which says "the sync to the given changelist is true and complete") from being created. This is communication among several concurrent processes.
